Fix KeyError on 'title' when title is empty

Mosibi opened the pull request:
When title is empty, a KeyError is thrown

---

ERROR:homeassistant.components.media_player:Error while setting up platform mpd
Traceback (most recent call last):
  File "/home/richard/home-assistant/homeassistant/helpers/entity_component.py", line 145, in _setup_platform
    self.hass, platform_config, self.add_entities, discovery_info)
  File "/home/richard/home-assistant/homeassistant/components/media_player/mpd.py", line 75, in setup_platform
    add_devices([MpdDevice(daemon, port, location, password)])
  File "/home/richard/home-assistant/homeassistant/helpers/entity_component.py", line 81, in add_entities
    entity.update_ha_state()
  File "/home/richard/home-assistant/homeassistant/helpers/entity.py", line 102, in update_ha_state
    attr = self.state_attributes or {}
  File "/home/richard/home-assistant/homeassistant/components/media_player/__init__.py", line 561, in state_attributes
    in ATTR_TO_PROPERTY if getattr(self, attr) is not None
  File "/home/richard/home-assistant/homeassistant/components/media_player/**init**.py", line 561, in <dictcomp>
    in ATTR_TO_PROPERTY if getattr(self, attr) is not None
  File "/home/richard/home-assistant/homeassistant/components/media_player/mpd.py", line 145, in media_title
    title = self.currentsong['title']
## KeyError: 'title'

Mosibi committed:
Fix KeyError on 'title' when title is empty

balloob commented on homeassistant/components/media_player/mpd.py:
```diff
@@ -142,12 +142,15 @@ def media_duration(self):
             return title
         else:
-            return '{}: {}'.format(name, title)
+            if title is None:
```
Can you collapse this if else check into the one above?

Mosibi replied:
I am not 100% sure that i understand you, but while looking at the code i noticed when 'name' is None, you would get something like 'None: $songtitle'. I will change my pull request so that it all the situations can be handled.

balloob replied:
Instead of
```
else:
    if title is None:
```

rewrite both if-blocks to be just `if X, elsif Y, elsif Z, else`

Mosibi committed:
Changed if else statements. The following situations are handled now:
  - name and title can be None
  - name can be None
  - title can be None
  - name and title can contain data

balloob commented:
CI reported two style violations.

Also, could you return None when neither title or name are available.

Mosibi committed:
Returning None when name and title are both not available
Removed trailing whitespaces

Mosibi committed:
identing error... sorry
